### ISSUE TITLE
didmovetowindow wrong behavior

### ISSUE THREAD

**Issue by alexburkhay** [John-Lluch/SWRevealTableViewCell#15]
Hello, i have found an issue:
Didmovetowindow in created cell calling every time we move from current screen or we move back to this screen. So for ex. we push some controller (it can be tableviewcontroller from your example) in our navigation stack :
        SWCellButtonItem *item2 = [SWCellButtonItem itemWithTitle:@"Open box" handler:^(SWCellButtonItem *item, SWRevealTableViewCell *cell)
        {
            _revealingCellIndexPath = [self.tableView indexPathForCell:cell];
            TableViewController *tableViewController = [[TableViewController alloc] initWithStyle:UITableViewStylePlain];
            [self.navigationController pushViewController:tableViewController animated:YES];
.....

so every time when current tableviewcontroller dissmised and appeared didmovetowindow called in every created cell and new utilityview created and added again to our cell in didmovetowindow:
        _utilityContentView = [[SWUtilityContentView alloc] initWithRevealTableViewCell:self frame:self.bounds];
        [_utilityContentView setAutoresizesSubviews:NO];
        [_utilityContentView setAutoresizingMask:UIViewAutoresizingFlexibleWidth|UIViewAutoresizingFlexibleHeight];
        [_revealLayoutView insertSubview:_utilityContentView atIndex:0];

So number of our contentviews increases infinitely:
po [_revealLayoutView subviews]
__NSArrayM 0x8d77150>(
   SWUtilityContentView: 0x8c70470; frame = (0 0; 320 44); autoresize = W+H;...,
  SWUtilityContentView: 0x9c8a170; frame = (0 0; 320 43.5); autoresize = W+H;...,
  SWUtilityContentView: 0x9c66d30; frame = (0 0; 320 43.5); autoresize = W+H;...,
  SWUtilityContentView: 0x8fb6040; frame = (0 0; 320 43.5); autoresize = W+H; ...,
  <SWUtilityContentView: 0x9863a30; frame = (0 0; 320 43.5); autoresize = W+H; ..,
  UITableViewCellContentView: 0x9861d30; frame = (0 0; 320 43.5); gestureRecognizers = ...,
  _UITableViewCellSeparatorView: 0x9863c90; frame = (47 43.5; 273 0.5); layer = ...
)

Also because of didmovetowindow 
        // Finally, set the actual position
        [self _setRevealPosition:initialPosition withDuration:0.0];
delegate methods like revealTableViewCell:(SWRevealTableViewCell *)revealTableViewCell didMoveToPosition: are called every time - issue in here is that if u implement logic when u can show again last opened cell after scrolling and reusing some created cell for this one (not implemented in current example - u can do this through adding ivar for ex. lastIndexPath and set value on didmovetoposition) and u want to hide that cell when user goes from current screen but cell not visible (but it exists) you only can set lastIndexPath to nil in viewdiddissapear, but after that delegate sets lastIndexPath to prev value of opened cell on didmovetowindow. 
Basically i want to say that on next didmovetowindow initialPosition equal SWCellRevealPositionLeft and all delegate methods called for unknown purpose.. so maybe we must use logic in didmovetowindow only one time per cell ?

**Comment by John-Lluch** [comment]
Hi @alexburkhay, you are absolutely right, this is definitely a bug. 

A quick fix for this is to call  `[_utilityContentView removeFromSuperview];` just before `_utilityContentView = [[SWUtilityContentView alloc] initWithRevealTableViewCell:self frame:self.bounds];`. This way, in case there was already a _utilityContentView around it will be replaced. However, I am looking for something more clever such as creating the utilityView lazily.

About your comment on the last paragraph I understand that delegates should not be called when the cell re-appears as this is inconsistent, but please can you clarify what is the effect of this on your code?

Thanks.

**Comment by John-Lluch** [comment]
Hi, I just pushed version v0.3.4. 

I think it should solve all your issues, please test it and let me know.

Thanks
